### PR TITLE
Add test case for namespaced exception catching

### PR DESCRIPTION
**Pull request by st0012**

### CODE REVIEW

File: test/debug/catch_test.rb [hunk 1]
```diff
@@ -63,4 +63,47 @@ def test_debugger_stops_when_the_exception_raised
...
+         9| end
+        10|
+        11| # we need this rescue + binding.bp workaround because the test framework can't handle exception exit yet
+        12| Foo.raised_exception rescue nil
+        13|
+        14| binding.bp
+      RUBY
+    end
+
+    def test_debugger_not_stopping_at_exception_without_namespace
```
Comment (ono-max) [inline test/debug/catch_test.rb:87]: How about this one?
```suggestion
    def test_catch_does_not_stop_at_exception_without_namespace
```

File: test/debug/catch_test.rb [hunk 2]
```diff
@@ -63,4 +63,47 @@ def test_debugger_stops_when_the_exception_raised
...
+      debug_code(program) do
+        type 'catch TestException'
+        type 'continue'
+        assert_line_num(14)
+        type 'quit'
+        type 'y'
+      end
+    end
+
+    def test_debugger_stop_at_exception_with_namespace
```
Comment (ono-max) [inline test/debug/catch_test.rb:97]: How about this one too?
```suggestion
    def test_catch_with_namespace_stops_at_exception
```

**Comment by ko1** [comment]
@st0012 how about suggestions?

**Comment by st0012** [comment]
@ko1 @ono-max sorry that I missed the suggestions. have applied them.

**Pull request merged**